### ISSUE TITLE
PathSeq Prealigned Host Filter

### ISSUE THREAD

**Issue by wir963** [broadinstitute/gatk#6687]
### Affected tool(s) or class(es)
PathSeq

### Affected version(s)
version 4.1.6.0

### Description 
I am pre-aligning a paired-end RNA-seq dataset to the human host genome using STAR. I get a `Uniquely mapped reads %` of 83.24%. Next, I run PathSeqPipelineSpark using `--is-host-aligned true` and the default options. When I look at the filter-metrics file, I see `PRIMARY_READS` = 1,057,098 and `READS_AFTER_PREALIGNED_HOST_FILTER` = 543,664

#### Steps to reproduce
This is on a dataset that I can't share but I'm happy to reproduce this on a publicly available dataset if you need that. 

#### Expected behavior
I would expect >= 83.24% of reads to be filtered by the PREALIGNED_HOST_FILTER step.  

#### Actual behavior
< 50% of the reads are filtered by the PREALIGNED_HOST_FILTER step. 

Do you have any insight into what's going on? Should I be using an aligner other than STAR? Are there RNA-seq specific instructions for running PathSeq?

**Comment by wir963** [comment]
After looking through the source code, I think it is relevant to add that this dataset uses 32 bp paired-end sequencing. I think the issue is that I am using the default `--host-min-identity 30` argument. Is the --host-min-identity for each mate or is it for both mates of a paired end read?

**Comment by mwalker174** [comment]
Hi @wir963, the min identity is for a single mate, and I agree that this is probably why reads are failing to be filtered for 32-base reads. I would try lowering it to around 20 and see if that helps.

**Comment by wir963** [comment]
Hey @mwalker174,

Setting `--host-min-identity 20` lowered the `READS_AFTER_PREALIGNED_HOST_FILTER` to 131467, which is closer to what I expected (albeit still 12% of the reads) so thanks for the suggestion.

I'm still unsure why the preferred approach is to use this fairly arbitrary metric that doesn't incorporate both mates instead of leveraging the known alignments from STAR, which does. Given what I've learned today, I think the better approach (for my use case) would be to filter the unique mappers, the multi-mappers and the chimeric reads (which in my data set represent 97.5% of the reads) and then apply the QUALITY_AND_COMPLEXITY_FILTER and the DUPLICATE_READS_FILTER. Would you agree?

To put myself in your shoes, I would guess that PathSeq is designed for general purpose use cases (which is probably `--is-host-aligned false`) and that performing such a filtering would require specific handling for each supported aligner, which would be a lot of work. Moreover, my use case with short paired-end reads is also probably not common. So I understand why you use the existing approach but are there any reasons that I'm missing as to why you'd suggest to stick with the basic approach for my use case instead of the one that I proposed above?

Best, Welles

**Comment by mwalker174** [comment]
Yes PathSeq has generally been used with longer reads in the past and also with BWA. I suspect with the shorter reads, novel splice sites are getting missed by PathSeq. You could turn down the match threshold to 16 (half the read length) to help this further, or even hard-filter any aligned read, but you do risk losing some microbial reads at this length. 

You're also right that in your case, using the mate information would be useful, but this would not be appropriate, for example, when there are host-pathogen chimeras, such as virus integration events.

**Comment by mwalker174** [comment]
@wir963 The [GATK forum](https://gatk.broadinstitute.org/hc/en-us/community/topics) is a good place for discussions like this.
